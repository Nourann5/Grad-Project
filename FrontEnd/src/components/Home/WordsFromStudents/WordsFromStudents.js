import React, { useState } from 'react';
import { Col, Container, Row } from 'react-bootstrap';
import Carousel from 'react-bootstrap/Carousel';
import styles from './WordsFromStudents.module.css'

import {ReactComponent as CommentsIcon} from 'assets/icons/comments.svg'
import {ReactComponent as QuoteDownIcon} from 'assets/icons/quoteDown.svg'
import {ReactComponent as QuoteTopIcon} from 'assets/icons/quoteTop.svg'
import {ReactComponent as LeftAnglePagIcon} from 'assets/icons/leftAnglePag.svg'
import {ReactComponent as RightAnglePagIcon} from 'assets/icons/rightAnglePag.svg'
import imgSrc from 'assets/icons/comments.png'

function WordsFromStudents() {
    const [index, setIndex] = useState(0);
    const handleSelect = (selectedIndex, e) => {
        setIndex(selectedIndex);
    };
  return (
    
    <div id={styles["comments"]}>
        <Container>
            <Row>
                <div className={`${styles["comments__content-cont"]} ${styles["comments__content-cont--responsive"]} d-none`}>
                    <h2 className={styles["comments__content-heading"]}>
                        WORDS FROM OUR STUDENTS
                    </h2>
                    <CommentsIcon className={styles["comments__content-heading-icon"]}/>
                </div>

                <Col lg='5' xs='12'>
                    <div className={`${styles["comments__img-cont"]} d-flex`}>
                        <img src={imgSrc} alt="comments" className={`${styles["comments__img"]} m-auto`}/>
                    </div>
                </Col>

                <Col lg='7' xs='12'>
                    <div className={styles["comments__content"]}>
                        <div className={styles["comments__content-cont"]}>
                            <h2 className={styles["comments__content-heading"]}>
                            WORDS<br/> FROM OUR Students
                            </h2>
                            <CommentsIcon className={styles["comments__content-heading-icon"]}/>

                        </div>
                        <div className={styles["comments__comments"]}>
                            <Carousel 
                            activeIndex={index} 
                            onSelect={handleSelect}
                            className={`${styles["carousel"]}`} 
                            indicators={false} >
                                <Carousel.Item className={`${styles["carousel-item"]}`}>
                                    
                                    <Carousel.Caption className={`${styles["carousel-caption"]}`}>
                                        <div className={styles["comments__comments-top-quote"]}>
                                            <QuoteTopIcon
                                            className={`${"comments__comments-quote"} ${"comments__comments-top-quote-icon"}`}/>
                                        </div>
                                        <p className={styles["comments__para"]}>
                                            Lorem ipsum dolor sit amet, consetetur sadipscing Lorem ipsum dolor sit amet, consetetur sadipscing Lorem ipsum dolor sit amet consetetur sadipscing Lorem ipsum
                                        </p>
                                        <div className={styles["comments__comments-foot"]}>
                                            <QuoteDownIcon
                                                className={`${"comments__comments-quote"} ${"comments__comments-down-quote-icon"}`}/>
                                            <p className={styles['comments__owner']}>
                                                Ahmed H. Ibrahim
                                            </p>
                                            <div className={styles["comments__comments-nav"]}>
                                                <button className={`${styles["carousel-control-prev"]}`} >
                                                    <LeftAnglePagIcon  className={styles["carousal__nav-icon"]} onClick={() => handleSelect('prev')}/>
                                                </button>
                                                <button className={`${styles["carousel-control-next"]}`} >
                                                    <RightAnglePagIcon className={styles["carousal__nav-icon"]} onClick={() => handleSelect('next')}/>
                                                </button>
                                            </div>

                                        </div>

                                    </Carousel.Caption>
                                </Carousel.Item>
                                <Carousel.Item className={`${styles["carousel-item"]}`}>
                                    
                                    <Carousel.Caption className={`${styles["carousel-caption"]}`}>
                                        <div className={styles["comments__comments-top-quote"]}>
                                            <QuoteTopIcon
                                            className={`${"comments__comments-quote"} ${"comments__comments-top-quote-icon"}`}/>
                                        </div>
                                        <p className={styles["comments__para"]}>
                                            Lorem ipsum dolor sit amet, consetetur sadipscing Lorem ipsum dolor sit amet, consetetur sadipscing Lorem ipsum dolor sit amet consetetur sadipscing Lorem ipsum
                                        </p>
                                        <div className={styles["comments__comments-foot"]}>
                                            <QuoteDownIcon
                                                className={`${"comments__comments-quote"} ${" comments__comments-down-quote-icon"}`}/>
                                            <p className={styles['comments__owner']}>
                                                Ahmed H. Ibrahim
                                            </p>
                                            <div className={styles["comments__comments-nav"]}>
                                            <button className={`${styles["carousel-control-prev"]}`} onClick={() => this.toggleCarousel('prev')}>
                                                    <LeftAnglePagIcon  className={styles["carousal__nav-icon"]}/>
                                                </button>
                                                <button className={`${styles["carousel-control-next"]}`} onClick={() => this.toggleCarousel('next')}>
                                                    <RightAnglePagIcon className={styles["carousal__nav-icon"]}/>
                                                </button>
                                            </div>

                                        </div>

                                    </Carousel.Caption>
                                </Carousel.Item>
                            </Carousel>
                        
                        </div>
                    </div>
                </Col>
            </Row>
        </Container>
    </div>
        
  );
}
export default WordsFromStudents