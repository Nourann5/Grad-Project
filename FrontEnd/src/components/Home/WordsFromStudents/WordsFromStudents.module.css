
#comments{
    padding:80px 0;
    background-color:var(--main-background-color)
}
#comments .carousel{
    min-height: 250px;
    transform: translate(-170px, 0px);
    border-radius: 15px;
    overflow: hidden;
}
.comments__content-cont--responsive{
    display:none
}
#comments .carousel-inner{
    height:100%;
    border-radius: 12px;
    background: white;
    box-shadow: 10px 10px 30px #4807E61A;
}
#comments .carousel-item{
    background: white;
    height: 100%;
    padding: 30px;
    border-radius: 12px;
}
#comments .carousel-caption{
    position: initial;
    color:var(--text-color-dark);
    padding:0;
}
.comments__comments-quote{
    width:40px;
    height: 40px;
}
.comments__comments-top-quote{
    text-align:right;
    margin-bottom:5px;
}
.comments__comments-down-quote-icon{
    margin-right: 20px;
}
#comments .carousel-control-next, #comments .carousel-control-prev{
    --size: 30px;
    position: initial;
    width: var(--size);
    height: var(--size);
    border: 2px solid var(--main-color)!important;
    border-radius: 50%;
    background-color: white;
    color: var(--main-color);
    cursor: pointer;
    opacity: 1;
}
#comments .carousel-control-next{
    left: 90%;
}
#comments .carousel-control-prev{
    left: 80%;
    margin-right: 10px;
}
.comments__para{
    font-family: 'PoppinsRegular';
    line-height: 30px;
    color: var(--text-color-dark);
    border-bottom: 1px solid #475677;
    padding-bottom: 30px;
    width: 95%;
    margin: auto;
    text-align: left;
}
.comments__comments-foot{
    display: flex;
    align-items:center;
    margin-top: 30px;
}
.comments__owner{
    margin: 0;
    font-size: 18px;
    font-family: 'PoppinsSemiBold';
    color:var(--text-color-dark);
}
.comments__comments-nav {
    display: flex;
    align-items: center;
    margin-left: auto;
}
.comments__content-heading{
    color: var(--text-color-dark);
    font-size: 35px;
    font-family: 'PlayFairBold';
}
.comments__content-cont{
    display: flex;
    align-items: center;
}
.comments__content-cont--responsive{
    display: none
}
.comments__content-heading-icon{
    margin-left: auto;
}
.comments__img{
    width: -webkit-fill-available;
}
.comments__content-heading-icon{
    width: 150px;
    height: 150px;
}
@media (max-width:1024px) {
    .comments__content-heading-icon{
        width: 120px;
        height: 120px;
    }
    .comments__content-heading{
        font-size:30px
    }
}
@media (max-width:992px) {
    .comments__content-heading{
        font-size:30px
    }
    .comments__img{
        width: 700px;
        height: 400px;
    }
    #comments .carousel{
        transform: translate(0px, -130px);
    }
    #comments .carousel-inner{
        width: 80%; 
        margin: auto;
    }

    .comments__content-cont{
        display: none;
    }
    .comments__content-cont--responsive{
        display: flex;
        width: 100%;
    }
    #comments{
        padding:0;
    }
}
@media (max-width:426px) {

    .comments__content-cont--responsive .comments__content-heading{
        font-size:25px
    }
    .comments__img{
        width: 400px;
        height: 300px;
    }
    #comments{
        padding:0 10px
    }
    #comments .carousel-inner{
        width:90%;
    }
    #comments .carousel-item{
        padding:15px
    }
    .comments__para{
        padding-bottom: 15px;
    }
    .comments__comments-foot{
        margin-top: 15px;
    }
    #comments .carousel-control-prev{
        margin-right: 5px;
    }
}
@media (max-width:376px) {
    .comments__img {
        width: 300px;
    }
}
@media (max-width:321px) {
    .comments__img {
        width: 250px;
    }
}