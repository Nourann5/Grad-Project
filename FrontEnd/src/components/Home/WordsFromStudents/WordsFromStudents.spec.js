import {render , screen} from '@testing-library/react'
import { BrowserRouter, Route } from 'react-router-dom'
import WordsFromStudents from './WordsFromStudents'

describe('Test the Home Words From Our Students', () => { 
    it('Heading exists',()=>{
        const tree =  render(
            <BrowserRouter>
                <WordsFromStudents/>
            </BrowserRouter>
        )
        const headerTitle = screen.getByRole('heading',{name:'WORDS FROM OUR STUDENTS'})
        expect(headerTitle).toBeInTheDocument()
        expect(tree).toMatchSnapshot()
    })
})