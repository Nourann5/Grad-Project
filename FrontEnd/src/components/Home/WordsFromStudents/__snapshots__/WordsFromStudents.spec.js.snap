// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Home Words From Our Students Heading exists 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="comments"
      >
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="comments__content-cont comments__content-cont--responsive d-none"
            >
              <h2
                class="comments__content-heading"
              >
                WORDS FROM OUR STUDENTS
              </h2>
              <svg
                class="comments__content-heading-icon"
              >
                comments.svg
              </svg>
            </div>
            <div
              class="col-lg-5 col-12"
            >
              <div
                class="comments__img-cont d-flex"
              >
                <img
                  alt="comments"
                  class="comments__img m-auto"
                  src="comments.png"
                />
              </div>
            </div>
            <div
              class="col-lg-7 col-12"
            >
              <div
                class="comments__content"
              >
                <div
                  class="comments__content-cont"
                >
                  <h2
                    class="comments__content-heading"
                  >
                    WORDS
                    <br />
                     FROM OUR Students
                  </h2>
                  <svg
                    class="comments__content-heading-icon"
                  >
                    comments.svg
                  </svg>
                </div>
                <div
                  class="comments__comments"
                >
                  <div
                    class="carousel carousel slide"
                  >
                    <div
                      class="carousel-inner"
                    >
                      <div
                        class="carousel-item active carousel-item"
                      >
                        <div
                          class="carousel-caption carousel-caption"
                        >
                          <div
                            class="comments__comments-top-quote"
                          >
                            <svg
                              class="comments__comments-quote comments__comments-top-quote-icon"
                            >
                              quoteTop.svg
                            </svg>
                          </div>
                          <p
                            class="comments__para"
                          >
                            Lorem ipsum dolor sit amet, consetetur sadipscing Lorem ipsum dolor sit amet, consetetur sadipscing Lorem ipsum dolor sit amet consetetur sadipscing Lorem ipsum
                          </p>
                          <div
                            class="comments__comments-foot"
                          >
                            <svg
                              class="comments__comments-quote comments__comments-down-quote-icon"
                            >
                              quoteDown.svg
                            </svg>
                            <p
                              class="comments__owner"
                            >
                              Ahmed H. Ibrahim
                            </p>
                            <div
                              class="comments__comments-nav"
                            >
                              <button
                                class="carousel-control-prev"
                              >
                                <svg
                                  class="carousal__nav-icon"
                                >
                                  leftAnglePag.svg
                                </svg>
                              </button>
                              <button
                                class="carousel-control-next"
                              >
                                <svg
                                  class="carousal__nav-icon"
                                >
                                  rightAnglePag.svg
                                </svg>
                              </button>
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="carousel-item carousel-item"
                      >
                        <div
                          class="carousel-caption carousel-caption"
                        >
                          <div
                            class="comments__comments-top-quote"
                          >
                            <svg
                              class="comments__comments-quote comments__comments-top-quote-icon"
                            >
                              quoteTop.svg
                            </svg>
                          </div>
                          <p
                            class="comments__para"
                          >
                            Lorem ipsum dolor sit amet, consetetur sadipscing Lorem ipsum dolor sit amet, consetetur sadipscing Lorem ipsum dolor sit amet consetetur sadipscing Lorem ipsum
                          </p>
                          <div
                            class="comments__comments-foot"
                          >
                            <svg
                              class="comments__comments-quote  comments__comments-down-quote-icon"
                            >
                              quoteDown.svg
                            </svg>
                            <p
                              class="comments__owner"
                            >
                              Ahmed H. Ibrahim
                            </p>
                            <div
                              class="comments__comments-nav"
                            >
                              <button
                                class="carousel-control-prev"
                              >
                                <svg
                                  class="carousal__nav-icon"
                                >
                                  leftAnglePag.svg
                                </svg>
                              </button>
                              <button
                                class="carousel-control-next"
                              >
                                <svg
                                  class="carousal__nav-icon"
                                >
                                  rightAnglePag.svg
                                </svg>
                              </button>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <a
                      class="carousel-control-prev"
                      href="#"
                      role="button"
                      tabindex="0"
                    >
                      <span
                        aria-hidden="true"
                        class="carousel-control-prev-icon"
                      />
                      <span
                        class="visually-hidden"
                      >
                        Previous
                      </span>
                    </a>
                    <a
                      class="carousel-control-next"
                      href="#"
                      role="button"
                      tabindex="0"
                    >
                      <span
                        aria-hidden="true"
                        class="carousel-control-next-icon"
                      />
                      <span
                        class="visually-hidden"
                      >
                        Next
                      </span>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="comments"
    >
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="comments__content-cont comments__content-cont--responsive d-none"
          >
            <h2
              class="comments__content-heading"
            >
              WORDS FROM OUR STUDENTS
            </h2>
            <svg
              class="comments__content-heading-icon"
            >
              comments.svg
            </svg>
          </div>
          <div
            class="col-lg-5 col-12"
          >
            <div
              class="comments__img-cont d-flex"
            >
              <img
                alt="comments"
                class="comments__img m-auto"
                src="comments.png"
              />
            </div>
          </div>
          <div
            class="col-lg-7 col-12"
          >
            <div
              class="comments__content"
            >
              <div
                class="comments__content-cont"
              >
                <h2
                  class="comments__content-heading"
                >
                  WORDS
                  <br />
                   FROM OUR Students
                </h2>
                <svg
                  class="comments__content-heading-icon"
                >
                  comments.svg
                </svg>
              </div>
              <div
                class="comments__comments"
              >
                <div
                  class="carousel carousel slide"
                >
                  <div
                    class="carousel-inner"
                  >
                    <div
                      class="carousel-item active carousel-item"
                    >
                      <div
                        class="carousel-caption carousel-caption"
                      >
                        <div
                          class="comments__comments-top-quote"
                        >
                          <svg
                            class="comments__comments-quote comments__comments-top-quote-icon"
                          >
                            quoteTop.svg
                          </svg>
                        </div>
                        <p
                          class="comments__para"
                        >
                          Lorem ipsum dolor sit amet, consetetur sadipscing Lorem ipsum dolor sit amet, consetetur sadipscing Lorem ipsum dolor sit amet consetetur sadipscing Lorem ipsum
                        </p>
                        <div
                          class="comments__comments-foot"
                        >
                          <svg
                            class="comments__comments-quote comments__comments-down-quote-icon"
                          >
                            quoteDown.svg
                          </svg>
                          <p
                            class="comments__owner"
                          >
                            Ahmed H. Ibrahim
                          </p>
                          <div
                            class="comments__comments-nav"
                          >
                            <button
                              class="carousel-control-prev"
                            >
                              <svg
                                class="carousal__nav-icon"
                              >
                                leftAnglePag.svg
                              </svg>
                            </button>
                            <button
                              class="carousel-control-next"
                            >
                              <svg
                                class="carousal__nav-icon"
                              >
                                rightAnglePag.svg
                              </svg>
                            </button>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="carousel-item carousel-item"
                    >
                      <div
                        class="carousel-caption carousel-caption"
                      >
                        <div
                          class="comments__comments-top-quote"
                        >
                          <svg
                            class="comments__comments-quote comments__comments-top-quote-icon"
                          >
                            quoteTop.svg
                          </svg>
                        </div>
                        <p
                          class="comments__para"
                        >
                          Lorem ipsum dolor sit amet, consetetur sadipscing Lorem ipsum dolor sit amet, consetetur sadipscing Lorem ipsum dolor sit amet consetetur sadipscing Lorem ipsum
                        </p>
                        <div
                          class="comments__comments-foot"
                        >
                          <svg
                            class="comments__comments-quote  comments__comments-down-quote-icon"
                          >
                            quoteDown.svg
                          </svg>
                          <p
                            class="comments__owner"
                          >
                            Ahmed H. Ibrahim
                          </p>
                          <div
                            class="comments__comments-nav"
                          >
                            <button
                              class="carousel-control-prev"
                            >
                              <svg
                                class="carousal__nav-icon"
                              >
                                leftAnglePag.svg
                              </svg>
                            </button>
                            <button
                              class="carousel-control-next"
                            >
                              <svg
                                class="carousal__nav-icon"
                              >
                                rightAnglePag.svg
                              </svg>
                            </button>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <a
                    class="carousel-control-prev"
                    href="#"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      aria-hidden="true"
                      class="carousel-control-prev-icon"
                    />
                    <span
                      class="visually-hidden"
                    >
                      Previous
                    </span>
                  </a>
                  <a
                    class="carousel-control-next"
                    href="#"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      aria-hidden="true"
                      class="carousel-control-next-icon"
                    />
                    <span
                      class="visually-hidden"
                    >
                      Next
                    </span>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
